src/main/java/core/basesyntax/FigureSupplierImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplierImpl implements FigureSupplier {
+    private Random random = new Random();
+
+    @Override
+    public Figure getRandomFigure() {
+        var shape = Shapes.values()[random.nextInt(Shapes.values().length)];
+        switch (shape) {
+            case RECTANGLE -> {
+                return new Rectangle(random.nextDouble(), random.nextDouble());
+            }
+            case SQUARE -> {
+                return new Square(random.nextDouble());
+            }
+            case CIRCLE -> {
+                return new Circle(random.nextDouble());
+            }
+            case RIGHT_TRIANGLE -> {
+                return new RightTriangle(random.nextDouble(), random.nextDouble());
+            }
+            case ISOSCELES_TRAPEZOID -> {
+                return new IsoscelesTrapezoid(
+                        random.nextDouble(),
+                        random.nextDouble(),
+                        random.nextDouble());
+            }
+            default -> {
+                return null;
```

Comment:
Returning `null` from a method is a bad practice. Consider handling this case differently, perhaps by returning a default figure or throwing an exception.